pages/index.js

```diff
@@ -1,48 +1,104 @@
 import AddCalories from "../components/AddCalories/AddCalories";
 import useCalorieStore from "../utils/useCalorieStore";
 import ConsumedList from "../components/ConsumedList/ConsumedList";
+import HomeCalendar from "../components/Calendar";
+import useLocalStorageState from "use-local-storage-state";
 import { useState, useEffect } from "react";
 import {
   StyledDiv,
   StyledButtonCalorieCounter,
-  ListContainer,
+  ConsumedContainer,
+  FormContainer,
+  OpenCalorieFormButton,
 } from "../components/IndexPage/styles";
 
 export default function HomePage() {
-  const { dailyCount, dailyMeals, calorieGoal } = useCalorieStore();
-  const difference = calorieGoal - dailyCount;
+  const { dailyCount, dailyMeals, calorieGoal, setDailyCount } =
+    useCalorieStore();
+
+  const [isLoading, setIsLoading] = useState(true);
 
   const [isListVisible, setIsListVisible] = useState(false);
   const [isGoalExceeded, setIsGoalExceeded] = useState(false);
+  const [isFormVisible, setIsFormVisible] = useState(false);
+
+  // reset the daily calorie count
+  const [currentDay, setCurrentDay] = useLocalStorageState("currentDay", {
+    defaultValue: new Date().getDay(),
+  });
+  setInterval(() => {
+    const today = new Date().getDay();
+    if (today !== currentDay) {
+      setDailyCount(0);
+      setCurrentDay(today);
+    }
+  }, 5000);
+  // <----------------------------------------
 
   useEffect(() => {
-    document.addEventListener("click", () => setIsListVisible(false));
+    document.addEventListener("click", () => {
```

Comment:
You should remove your event listener on unmount by returning a cleanup function. 
[How to clean up event listeners](https://stackoverflow.com/questions/61444561/react-hooks-how-to-clean-event-listeners)